src/java/org/apache/cassandra/cql3/UpdateParameters.java

```diff
@@ -123,18 +121,30 @@ public Clustering<?> currentClustering()
 
     public void addPrimaryKeyLivenessInfo()
     {
-        builder.addPrimaryKeyLivenessInfo(LivenessInfo.create(timestamp, ttl, nowInSec));
+        addPrimaryKeyLivenessInfo(LivenessInfo.create(timestamp, ttl, nowInSec));
+    }
+
+    //TODO hide?
+    public void addPrimaryKeyLivenessInfo(LivenessInfo info)
```

Comment:
don't like these but needed for TxnWrite... maybe its best to have something extend UP to expose this?